vllm/v1/core/sched/scheduler.py

```diff
@@ -1128,7 +1214,11 @@ def update_from_output(
                 stopped = check_stop(request, self.max_model_len, pooler_output)
 
             if stopped:
-                kv_transfer_params = self._free_request(request)
+                if request.resumable:
+                    request.status = RequestStatus.WAITING_FOR_STREAMING_REQ
+                    self.waiting.add_request(request)
```

Comment:
Here I think it might make sense to check whether there is another request in the streaming queue and "update the session" here if so (leaving the request state as `WAITING` rather than `WAITING_FOR_STREAMING_REQ`).

This might avoid the performance issue mentioned by @vklimkov-nvidia above:
> I noticed that for concurrency=1, the main scheduler loop may go to sleep waiting for messages in queue, which can be somewhat slow (acquiring lock takes time).